tests/unit/test_charm.py

```diff
@@ -570,6 +568,35 @@ def test_missing_pebble_plan(self, mock_validate_pebble_plan):
         plan = harness.get_container_pebble_plan("temporal").to_dict()
         assert plan is not None
 
+    def test_rendering(self):
+        """The dynamic config gets rendered correctly."""
+        expected_output = """frontend.namespacerps:
```

Comment:
suggestion: I like `text.dedent` and omitting new lines via `"""\` to make multi-line string indentation consistent with the code. But it is purely cosmetic.